From 4823fb6eb7f5e7899226074b8f00c4ca533ff25a Mon Sep 17 00:00:00 2001
From: Kevin Phoenix <kevin@kphoenix.us>
Date: Wed, 30 Aug 2023 15:25:38 -0700
Subject: [PATCH] Rewrite test_tmsr and test_tmsr to not use angr (#345)

---
 tests/test_spotter.py | 22 ++++++----------------
 1 file changed, 6 insertions(+), 16 deletions(-)

diff --git a/tests/test_spotter.py b/tests/test_spotter.py
index 9271ccd..2bf34f6 100644
--- a/tests/test_spotter.py
+++ b/tests/test_spotter.py
@@ -113,32 +113,22 @@ def test_full_binary():
 
 
 def test_tmrs():
-    test_location = str(os.path.join(os.path.dirname(os.path.realpath(__file__)), "../../binaries/tests"))
-    p = angr.Project(
-        os.path.join(test_location, "armel", "helloworld"),
-        arch="ARMEL",
-        auto_load_libs=False,
-    )
+    arch = archinfo.arch_from_id("ARMEL")
     ins = b"\xef\xf3\x08\x82"
-    b = pyvex.block.IRSB(ins, 1, p.arch)
+    b = pyvex.block.IRSB(ins, 1, arch)
     assert b.jumpkind == "Ijk_Boring"
     assert type(b.statements[1].data) == pyvex.expr.Get
-    assert p.arch.register_names.get(b.statements[1].data.offset, "") == "sp"
+    assert arch.register_names.get(b.statements[1].data.offset, "") == "sp"
     assert type(b.statements[2]) == pyvex.stmt.Put
 
 
 def test_tmsr():
-    test_location = str(os.path.join(os.path.dirname(os.path.realpath(__file__)), "../../binaries/tests"))
-    p = angr.Project(
-        os.path.join(test_location, "armel", "helloworld"),
-        arch="ARMEL",
-        auto_load_libs=False,
-    )
+    arch = archinfo.arch_from_id("ARMEL")
     inss = b"\x82\xf3\x08\x88"
-    b = pyvex.block.IRSB(inss, 1, p.arch, opt_level=3)
+    b = pyvex.block.IRSB(inss, 1, arch, opt_level=3)
     assert b.jumpkind == "Ijk_Boring"
     assert type(b.statements[1].data) == pyvex.expr.Get
-    assert p.arch.register_names.get(b.statements[1].data.offset, "") == "r2"
+    assert arch.register_names.get(b.statements[1].data.offset, "") == "r2"
     assert type(b.statements[2]) == pyvex.stmt.Put
 
 
