From b977ea62c922ba1bcc15dfc562afa50d4c07154c Mon Sep 17 00:00:00 2001
From: Kevin Phoenix <kevin@kphoenix.us>
Date: Thu, 31 Aug 2023 16:11:27 -0700
Subject: [PATCH] Remove full binary test (#346)

---
 tests/test_spotter.py | 16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/tests/test_spotter.py b/tests/test_spotter.py
index 2bf34f6..bed7dd4 100644
--- a/tests/test_spotter.py
+++ b/tests/test_spotter.py
@@ -1,6 +1,5 @@
 import os
 
-import angr
 import archinfo
 
 import pyvex
@@ -98,20 +97,6 @@ class CortexSpotter(GymratLifter):
 register(CortexSpotter, "ARMEL")
 
 
-def test_full_binary():
-    p = angr.Project(
-        os.path.join(test_location, "armel", "RTOSDemo.axf.issue_685"),
-        arch="ARMEL",
-        auto_load_libs=False,
-    )
-    st = p.factory.call_state(0x000013CE + 1)
-    b = st.block().vex
-    simgr = p.factory.simulation_manager(st)
-    simgr.step()
-    assert b.jumpkind == "Ijk_Sys_syscall"
-    assert simgr.active[0].regs.ip_at_syscall.args[0] == 0x13FB
-
-
 def test_tmrs():
     arch = archinfo.arch_from_id("ARMEL")
     ins = b"\xef\xf3\x08\x82"
@@ -135,6 +120,5 @@ def test_tmsr():
 if __name__ == "__main__":
     test_basic()
     test_embedded()
-    test_full_binary()
     test_tmrs()
     test_tmsr()
